diff --git a/concrete/access/ttypes.py b/concrete/access/ttypes.py
index 57fabed..0a244f6 100644
--- a/concrete/access/ttypes.py
+++ b/concrete/access/ttypes.py
@@ -8,6 +8,7 @@
 #
 
 from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol import TProtocol
 from thrift.protocol.TProtocol import TProtocolException
 import sys
 import concrete.communication.ttypes
diff --git a/concrete/annotate/ttypes.py b/concrete/annotate/ttypes.py
index 1ebbafb..58de779 100644
--- a/concrete/annotate/ttypes.py
+++ b/concrete/annotate/ttypes.py
@@ -8,6 +8,7 @@
 #
 
 from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol import TProtocol
 from thrift.protocol.TProtocol import TProtocolException
 import sys
 import concrete.metadata.ttypes
diff --git a/concrete/audio/ttypes.py b/concrete/audio/ttypes.py
index faa7b28..67a9036 100644
--- a/concrete/audio/ttypes.py
+++ b/concrete/audio/ttypes.py
@@ -8,6 +8,7 @@
 #
 
 from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol import TProtocol
 from thrift.protocol.TProtocol import TProtocolException
 import sys
 
diff --git a/concrete/clustering/ttypes.py b/concrete/clustering/ttypes.py
index 46be9fa..8de0496 100644
--- a/concrete/clustering/ttypes.py
+++ b/concrete/clustering/ttypes.py
@@ -8,6 +8,7 @@
 #
 
 from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol import TProtocol
 from thrift.protocol.TProtocol import TProtocolException
 import sys
 import concrete.uuid.ttypes
diff --git a/concrete/communication/ttypes.py b/concrete/communication/ttypes.py
index 333b183..1a9a17c 100644
--- a/concrete/communication/ttypes.py
+++ b/concrete/communication/ttypes.py
@@ -8,6 +8,7 @@
 #
 
 from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol import TProtocol
 from thrift.protocol.TProtocol import TProtocolException
 import sys
 import concrete.uuid.ttypes
diff --git a/concrete/email/ttypes.py b/concrete/email/ttypes.py
index 0025d8d..6062ca4 100644
--- a/concrete/email/ttypes.py
+++ b/concrete/email/ttypes.py
@@ -8,6 +8,7 @@
 #
 
 from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol import TProtocol
 from thrift.protocol.TProtocol import TProtocolException
 import sys
 
diff --git a/concrete/entities/ttypes.py b/concrete/entities/ttypes.py
index 474d2b4..5600f13 100644
--- a/concrete/entities/ttypes.py
+++ b/concrete/entities/ttypes.py
@@ -8,6 +8,7 @@
 #
 
 from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol import TProtocol
 from thrift.protocol.TProtocol import TProtocolException
 import sys
 import concrete.structure.ttypes
diff --git a/concrete/exceptions/ttypes.py b/concrete/exceptions/ttypes.py
index 72eaf9f..49a902a 100644
--- a/concrete/exceptions/ttypes.py
+++ b/concrete/exceptions/ttypes.py
@@ -8,6 +8,7 @@
 #
 
 from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol import TProtocol
 from thrift.protocol.TProtocol import TProtocolException
 import sys
 
diff --git a/concrete/language/ttypes.py b/concrete/language/ttypes.py
index d20ec95..e8a9088 100644
--- a/concrete/language/ttypes.py
+++ b/concrete/language/ttypes.py
@@ -8,6 +8,7 @@
 #
 
 from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol import TProtocol
 from thrift.protocol.TProtocol import TProtocolException
 import sys
 import concrete.metadata.ttypes
diff --git a/concrete/learn/ttypes.py b/concrete/learn/ttypes.py
index debd780..f3db903 100644
--- a/concrete/learn/ttypes.py
+++ b/concrete/learn/ttypes.py
@@ -8,6 +8,7 @@
 #
 
 from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol import TProtocol
 from thrift.protocol.TProtocol import TProtocolException
 import sys
 import concrete.services.ttypes
diff --git a/concrete/linking/ttypes.py b/concrete/linking/ttypes.py
index a128872..c8de437 100644
--- a/concrete/linking/ttypes.py
+++ b/concrete/linking/ttypes.py
@@ -8,6 +8,7 @@
 #
 
 from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol import TProtocol
 from thrift.protocol.TProtocol import TProtocolException
 import sys
 import concrete.uuid.ttypes
diff --git a/concrete/metadata/ttypes.py b/concrete/metadata/ttypes.py
index 81885a3..faa4d77 100644
--- a/concrete/metadata/ttypes.py
+++ b/concrete/metadata/ttypes.py
@@ -8,6 +8,7 @@
 #
 
 from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol import TProtocol
 from thrift.protocol.TProtocol import TProtocolException
 import sys
 import concrete.uuid.ttypes
diff --git a/concrete/nitf/ttypes.py b/concrete/nitf/ttypes.py
index 93bf6b0..a102270 100644
--- a/concrete/nitf/ttypes.py
+++ b/concrete/nitf/ttypes.py
@@ -8,6 +8,7 @@
 #
 
 from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol import TProtocol
 from thrift.protocol.TProtocol import TProtocolException
 import sys
 
diff --git a/concrete/search/ttypes.py b/concrete/search/ttypes.py
index 56290da..386ff7e 100644
--- a/concrete/search/ttypes.py
+++ b/concrete/search/ttypes.py
@@ -8,6 +8,7 @@
 #
 
 from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol import TProtocol
 from thrift.protocol.TProtocol import TProtocolException
 import sys
 import concrete.services.ttypes
diff --git a/concrete/services/results/ttypes.py b/concrete/services/results/ttypes.py
index 3e662b0..7c5b3b8 100644
--- a/concrete/services/results/ttypes.py
+++ b/concrete/services/results/ttypes.py
@@ -8,6 +8,7 @@
 #
 
 from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol import TProtocol
 from thrift.protocol.TProtocol import TProtocolException
 import sys
 import concrete.services.ttypes
diff --git a/concrete/services/ttypes.py b/concrete/services/ttypes.py
index 8545779..76e2ec0 100644
--- a/concrete/services/ttypes.py
+++ b/concrete/services/ttypes.py
@@ -8,6 +8,7 @@
 #
 
 from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol import TProtocol
 from thrift.protocol.TProtocol import TProtocolException
 import sys
 import concrete.uuid.ttypes
diff --git a/concrete/situations/ttypes.py b/concrete/situations/ttypes.py
index f052a89..8a77ddc 100644
--- a/concrete/situations/ttypes.py
+++ b/concrete/situations/ttypes.py
@@ -8,6 +8,7 @@
 #
 
 from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol import TProtocol
 from thrift.protocol.TProtocol import TProtocolException
 import sys
 import concrete.structure.ttypes
diff --git a/concrete/spans/ttypes.py b/concrete/spans/ttypes.py
index 79bad0b..bfe6c47 100644
--- a/concrete/spans/ttypes.py
+++ b/concrete/spans/ttypes.py
@@ -8,6 +8,7 @@
 #
 
 from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol import TProtocol
 from thrift.protocol.TProtocol import TProtocolException
 import sys
 
diff --git a/concrete/structure/ttypes.py b/concrete/structure/ttypes.py
index 3111705..acb9847 100644
--- a/concrete/structure/ttypes.py
+++ b/concrete/structure/ttypes.py
@@ -8,6 +8,7 @@
 #
 
 from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol import TProtocol
 from thrift.protocol.TProtocol import TProtocolException
 import sys
 import concrete.metadata.ttypes
@@ -1993,8 +1994,14 @@ class Tokenization(object):
         return
 
     def __repr__(self):
+        # The implementation of this function is NOT auto-generated
+        # Prevent infinite recursion caused by backpointer added by
+        #   concrete.util.references.add_references_to_communication()
+        dict_without_backpointer = self.__dict__.copy()
+        if 'sentence' in dict_without_backpointer:
+            dict_without_backpointer.pop('sentence')
         L = ['%s=%r' % (key, value)
-             for key, value in self.__dict__.items()]
+             for key, value in dict_without_backpointer.iteritems()]
         return '%s(%s)' % (self.__class__.__name__, ', '.join(L))
 
     def __eq__(self, other):
diff --git a/concrete/twitter/ttypes.py b/concrete/twitter/ttypes.py
index 7238fb7..d4d8de4 100644
--- a/concrete/twitter/ttypes.py
+++ b/concrete/twitter/ttypes.py
@@ -8,6 +8,7 @@
 #
 
 from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol import TProtocol
 from thrift.protocol.TProtocol import TProtocolException
 import sys
 
diff --git a/concrete/uuid/ttypes.py b/concrete/uuid/ttypes.py
index 58cce83..f1143d6 100644
--- a/concrete/uuid/ttypes.py
+++ b/concrete/uuid/ttypes.py
@@ -8,6 +8,7 @@
 #
 
 from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol import TProtocol
 from thrift.protocol.TProtocol import TProtocolException
 import sys

