diff -ruN -x '*.pyc' a/email/ttypes.py b/email/ttypes.py
--- a/email/ttypes.py	2015-11-18 21:25:52.467371486 +0000
+++ b/email/ttypes.py	2015-11-18 21:25:41.303384377 +0000
@@ -23,7 +23,7 @@
   values are typically extracted from strings such as:
   <tt> "John Smith" &lt;john\@xyz.com&gt; </tt>.
 
-  \see RFC2822 http://tools.ietf.org/html/rfc2822
+  \see RFC2822 <http://tools.ietf.org/html/rfc2822>
 
   Attributes:
    - address
