diff -ruN -x '*.pyc' a/entities/ttypes.py b/entities/ttypes.py
--- a/entities/ttypes.py	2015-09-28 14:13:02.786092000 -0400
+++ b/entities/ttypes.py	2015-09-27 09:07:18.415379000 -0400
@@ -147,6 +147,15 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.uuid)
+    value = (value * 31) ^ hash(self.mentionIdList)
+    value = (value * 31) ^ hash(self.type)
+    value = (value * 31) ^ hash(self.confidence)
+    value = (value * 31) ^ hash(self.canonicalName)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -289,6 +298,15 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.uuid)
+    value = (value * 31) ^ hash(self.metadata)
+    value = (value * 31) ^ hash(self.entityList)
+    value = (value * 31) ^ hash(self.linkingList)
+    value = (value * 31) ^ hash(self.mentionSetId)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -467,6 +485,17 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.uuid)
+    value = (value * 31) ^ hash(self.tokens)
+    value = (value * 31) ^ hash(self.entityType)
+    value = (value * 31) ^ hash(self.phraseType)
+    value = (value * 31) ^ hash(self.confidence)
+    value = (value * 31) ^ hash(self.text)
+    value = (value * 31) ^ hash(self.childMentionIdList)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -596,6 +625,14 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.uuid)
+    value = (value * 31) ^ hash(self.metadata)
+    value = (value * 31) ^ hash(self.mentionList)
+    value = (value * 31) ^ hash(self.linkingList)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
