diff -ruN -x '*.pyc' a/metadata/ttypes.py b/metadata/ttypes.py
--- a/metadata/ttypes.py	2015-11-18 21:25:52.439371512 +0000
+++ b/metadata/ttypes.py	2015-11-18 21:25:41.303384377 +0000
@@ -756,7 +756,7 @@
    - tweetInfo: Extra information for communications where kind==TWEET:
   Information about this tweet that is provided by the Twitter
   API.  For information about the Twitter API, see:
-  https://dev.twitter.com/docs/platform-objects
+  <https://dev.twitter.com/docs/platform-objects>
    - emailInfo: Extra information for communications where kind==EMAIL
    - nitfInfo: Extra information that may come from the NITF
   (News Industry Text Format) schema. See 'nitf.thrift'.
