diff -ruN -x '*.pyc' a/situations/ttypes.py b/situations/ttypes.py
--- a/situations/ttypes.py	2015-09-28 14:13:02.877806000 -0400
+++ b/situations/ttypes.py	2015-09-27 09:07:18.499447000 -0400
@@ -107,6 +107,13 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.value)
+    value = (value * 31) ^ hash(self.metadata)
+    value = (value * 31) ^ hash(self.polarity)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -223,6 +230,14 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.role)
+    value = (value * 31) ^ hash(self.entityId)
+    value = (value * 31) ^ hash(self.situationId)
+    value = (value * 31) ^ hash(self.propertyList)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -325,6 +340,13 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.justificationType)
+    value = (value * 31) ^ hash(self.mentionId)
+    value = (value * 31) ^ hash(self.tokenRefSeqList)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -411,6 +433,13 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.timeMLClass)
+    value = (value * 31) ^ hash(self.timeMLTense)
+    value = (value * 31) ^ hash(self.timeMLAspect)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -842,6 +871,20 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.uuid)
+    value = (value * 31) ^ hash(self.situationType)
+    value = (value * 31) ^ hash(self.situationKind)
+    value = (value * 31) ^ hash(self.argumentList)
+    value = (value * 31) ^ hash(self.mentionIdList)
+    value = (value * 31) ^ hash(self.justificationList)
+    value = (value * 31) ^ hash(self.timeML)
+    value = (value * 31) ^ hash(self.intensity)
+    value = (value * 31) ^ hash(self.polarity)
+    value = (value * 31) ^ hash(self.confidence)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -967,6 +1010,14 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.uuid)
+    value = (value * 31) ^ hash(self.metadata)
+    value = (value * 31) ^ hash(self.situationList)
+    value = (value * 31) ^ hash(self.linkingList)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -1124,6 +1175,17 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.role)
+    value = (value * 31) ^ hash(self.entityMentionId)
+    value = (value * 31) ^ hash(self.situationMentionId)
+    value = (value * 31) ^ hash(self.tokens)
+    value = (value * 31) ^ hash(self.confidence)
+    value = (value * 31) ^ hash(self.propertyList)
+    value = (value * 31) ^ hash(self.constituent)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -1538,6 +1600,20 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.uuid)
+    value = (value * 31) ^ hash(self.text)
+    value = (value * 31) ^ hash(self.situationType)
+    value = (value * 31) ^ hash(self.situationKind)
+    value = (value * 31) ^ hash(self.argumentList)
+    value = (value * 31) ^ hash(self.intensity)
+    value = (value * 31) ^ hash(self.polarity)
+    value = (value * 31) ^ hash(self.tokens)
+    value = (value * 31) ^ hash(self.constituent)
+    value = (value * 31) ^ hash(self.confidence)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -1663,6 +1739,14 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.uuid)
+    value = (value * 31) ^ hash(self.metadata)
+    value = (value * 31) ^ hash(self.mentionList)
+    value = (value * 31) ^ hash(self.linkingList)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
