--- a/concrete/structure/ttypes.py
+++ b/concrete/structure/ttypes.py
@@ -1916,8 +1916,14 @@ class Tokenization(object):
         return

     def __repr__(self):
+        # The implementation of this function is NOT auto-generated
+        # Prevent infinite recursion caused by backpointer added by
+        #   concrete.util.references.add_references_to_communication()
+        dict_without_backpointer = self.__dict__.copy()
+        if 'sentence' in dict_without_backpointer:
+            dict_without_backpointer.pop('sentence')
         L = ['%s=%r' % (key, value)
-             for key, value in self.__dict__.items()]
+             for key, value in dict_without_backpointer.items()]
         return '%s(%s)' % (self.__class__.__name__, ', '.join(L))

     def __eq__(self, other):
