diff -ruN -x '*.pyc' a/twitter/ttypes.py b/twitter/ttypes.py
--- a/twitter/ttypes.py	2015-09-27 09:07:16.767483000 -0400
+++ b/twitter/ttypes.py	2015-09-27 09:00:42.364587000 -0400
@@ -262,6 +262,27 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.id)
+    value = (value * 31) ^ hash(self.name)
+    value = (value * 31) ^ hash(self.screenName)
+    value = (value * 31) ^ hash(self.lang)
+    value = (value * 31) ^ hash(self.geoEnabled)
+    value = (value * 31) ^ hash(self.createdAt)
+    value = (value * 31) ^ hash(self.friendsCount)
+    value = (value * 31) ^ hash(self.statusesCount)
+    value = (value * 31) ^ hash(self.verified)
+    value = (value * 31) ^ hash(self.listedCount)
+    value = (value * 31) ^ hash(self.favouritesCount)
+    value = (value * 31) ^ hash(self.followersCount)
+    value = (value * 31) ^ hash(self.location)
+    value = (value * 31) ^ hash(self.timeZone)
+    value = (value * 31) ^ hash(self.description)
+    value = (value * 31) ^ hash(self.utcOffset)
+    value = (value * 31) ^ hash(self.url)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -336,6 +357,12 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.latitude)
+    value = (value * 31) ^ hash(self.longitude)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -419,6 +446,12 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.type)
+    value = (value * 31) ^ hash(self.coordinateList)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -503,6 +536,13 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.streetAddress)
+    value = (value * 31) ^ hash(self.region)
+    value = (value * 31) ^ hash(self.locality)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -612,6 +652,15 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.startOffset)
+    value = (value * 31) ^ hash(self.endOffset)
+    value = (value * 31) ^ hash(self.screenName)
+    value = (value * 31) ^ hash(self.name)
+    value = (value * 31) ^ hash(self.id)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -720,6 +769,15 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.startOffset)
+    value = (value * 31) ^ hash(self.endOffset)
+    value = (value * 31) ^ hash(self.expandedUrl)
+    value = (value * 31) ^ hash(self.url)
+    value = (value * 31) ^ hash(self.displayUrl)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -804,6 +862,13 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.text)
+    value = (value * 31) ^ hash(self.startOffset)
+    value = (value * 31) ^ hash(self.endOffset)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -915,6 +980,13 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.hashtagList)
+    value = (value * 31) ^ hash(self.urlList)
+    value = (value * 31) ^ hash(self.userMentionList)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -1073,6 +1145,19 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.placeType)
+    value = (value * 31) ^ hash(self.countryCode)
+    value = (value * 31) ^ hash(self.country)
+    value = (value * 31) ^ hash(self.fullName)
+    value = (value * 31) ^ hash(self.name)
+    value = (value * 31) ^ hash(self.id)
+    value = (value * 31) ^ hash(self.url)
+    value = (value * 31) ^ hash(self.boundingBox)
+    value = (value * 31) ^ hash(self.attributes)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -1146,6 +1231,12 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.type)
+    value = (value * 31) ^ hash(self.coordinates)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -1381,6 +1472,25 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.id)
+    value = (value * 31) ^ hash(self.text)
+    value = (value * 31) ^ hash(self.createdAt)
+    value = (value * 31) ^ hash(self.user)
+    value = (value * 31) ^ hash(self.truncated)
+    value = (value * 31) ^ hash(self.entities)
+    value = (value * 31) ^ hash(self.source)
+    value = (value * 31) ^ hash(self.coordinates)
+    value = (value * 31) ^ hash(self.place)
+    value = (value * 31) ^ hash(self.favorited)
+    value = (value * 31) ^ hash(self.retweeted)
+    value = (value * 31) ^ hash(self.retweetCount)
+    value = (value * 31) ^ hash(self.inReplyToScreenName)
+    value = (value * 31) ^ hash(self.inReplyToStatusId)
+    value = (value * 31) ^ hash(self.inReplyToUserId)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
