diff -ruN -x '*.pyc' a/uuid/ttypes.py b/uuid/ttypes.py
--- a/uuid/ttypes.py	2015-09-28 14:13:02.942815000 -0400
+++ b/uuid/ttypes.py	2015-09-27 09:07:18.538020000 -0400
@@ -71,6 +71,11 @@
     return
 
 
+  def __hash__(self):
+    value = 17
+    value = (value * 31) ^ hash(self.uuidString)
+    return value
+
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
