diff -ru a/concrete/audio/ttypes.py b/concrete/audio/ttypes.py
--- a/concrete/audio/ttypes.py	2015-09-26 07:59:50.897461000 -0400
+++ b/concrete/audio/ttypes.py	2015-09-26 07:29:58.592006000 -0400
@@ -112,14 +112,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.wav)
-    value = (value * 31) ^ hash(self.mp3)
-    value = (value * 31) ^ hash(self.sph)
-    value = (value * 31) ^ hash(self.path)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
diff -ru a/concrete/clustering/ttypes.py b/concrete/clustering/ttypes.py
--- a/concrete/clustering/ttypes.py	2015-09-26 07:59:51.214061000 -0400
+++ b/concrete/clustering/ttypes.py	2015-09-26 07:29:58.611563000 -0400
@@ -110,13 +110,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.communicationId)
-    value = (value * 31) ^ hash(self.setId)
-    value = (value * 31) ^ hash(self.elementId)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -232,13 +225,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.clusterMemberIndexList)
-    value = (value * 31) ^ hash(self.confidenceList)
-    value = (value * 31) ^ hash(self.childIndexList)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -383,15 +369,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.uuid)
-    value = (value * 31) ^ hash(self.metadata)
-    value = (value * 31) ^ hash(self.clusterMemberList)
-    value = (value * 31) ^ hash(self.clusterList)
-    value = (value * 31) ^ hash(self.rootClusterIndexList)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
diff -ru a/concrete/communication/ttypes.py b/concrete/communication/ttypes.py
--- a/concrete/communication/ttypes.py	2015-09-26 07:59:51.034863000 -0400
+++ b/concrete/communication/ttypes.py	2015-09-26 07:29:58.708399000 -0400
@@ -158,15 +158,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.uuid)
-    value = (value * 31) ^ hash(self.metadata)
-    value = (value * 31) ^ hash(self.taggingType)
-    value = (value * 31) ^ hash(self.tagList)
-    value = (value * 31) ^ hash(self.confidenceList)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -565,28 +556,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.id)
-    value = (value * 31) ^ hash(self.uuid)
-    value = (value * 31) ^ hash(self.type)
-    value = (value * 31) ^ hash(self.text)
-    value = (value * 31) ^ hash(self.startTime)
-    value = (value * 31) ^ hash(self.endTime)
-    value = (value * 31) ^ hash(self.communicationTaggingList)
-    value = (value * 31) ^ hash(self.metadata)
-    value = (value * 31) ^ hash(self.keyValueMap)
-    value = (value * 31) ^ hash(self.lidList)
-    value = (value * 31) ^ hash(self.sectionList)
-    value = (value * 31) ^ hash(self.entityMentionSetList)
-    value = (value * 31) ^ hash(self.entitySetList)
-    value = (value * 31) ^ hash(self.situationMentionSetList)
-    value = (value * 31) ^ hash(self.situationSetList)
-    value = (value * 31) ^ hash(self.originalText)
-    value = (value * 31) ^ hash(self.sound)
-    value = (value * 31) ^ hash(self.communicationMetadata)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -769,16 +738,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.communicationIdList)
-    value = (value * 31) ^ hash(self.corpus)
-    value = (value * 31) ^ hash(self.entityMentionClusterList)
-    value = (value * 31) ^ hash(self.entityClusterList)
-    value = (value * 31) ^ hash(self.situationMentionClusterList)
-    value = (value * 31) ^ hash(self.situationClusterList)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
Only in a/concrete: discourse
diff -ru a/concrete/email/ttypes.py b/concrete/email/ttypes.py
--- a/concrete/email/ttypes.py	2015-09-26 07:59:50.829294000 -0400
+++ b/concrete/email/ttypes.py	2015-09-26 07:29:58.744761000 -0400
@@ -23,7 +23,7 @@
   values are typically extracted from strings such as:
   <tt> "John Smith" &lt;john\@xyz.com&gt; </tt>.
 
-  \see RFC2822 <http://tools.ietf.org/html/rfc2822>
+  \see RFC2822 http://tools.ietf.org/html/rfc2822
 
   Attributes:
    - address
@@ -84,12 +84,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.address)
-    value = (value * 31) ^ hash(self.displayName)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -393,26 +387,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.messageId)
-    value = (value * 31) ^ hash(self.contentType)
-    value = (value * 31) ^ hash(self.userAgent)
-    value = (value * 31) ^ hash(self.inReplyToList)
-    value = (value * 31) ^ hash(self.referenceList)
-    value = (value * 31) ^ hash(self.senderAddress)
-    value = (value * 31) ^ hash(self.returnPathAddress)
-    value = (value * 31) ^ hash(self.toAddressList)
-    value = (value * 31) ^ hash(self.ccAddressList)
-    value = (value * 31) ^ hash(self.bccAddressList)
-    value = (value * 31) ^ hash(self.emailFolder)
-    value = (value * 31) ^ hash(self.subject)
-    value = (value * 31) ^ hash(self.quotedAddresses)
-    value = (value * 31) ^ hash(self.attachmentPaths)
-    value = (value * 31) ^ hash(self.salutation)
-    value = (value * 31) ^ hash(self.signature)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
diff -ru a/concrete/entities/ttypes.py b/concrete/entities/ttypes.py
--- a/concrete/entities/ttypes.py	2015-09-26 07:59:50.663209000 -0400
+++ b/concrete/entities/ttypes.py	2015-09-26 07:29:58.792012000 -0400
@@ -147,15 +147,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.uuid)
-    value = (value * 31) ^ hash(self.mentionIdList)
-    value = (value * 31) ^ hash(self.type)
-    value = (value * 31) ^ hash(self.confidence)
-    value = (value * 31) ^ hash(self.canonicalName)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -298,15 +289,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.uuid)
-    value = (value * 31) ^ hash(self.metadata)
-    value = (value * 31) ^ hash(self.entityList)
-    value = (value * 31) ^ hash(self.linkingList)
-    value = (value * 31) ^ hash(self.mentionSetId)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -485,17 +467,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.uuid)
-    value = (value * 31) ^ hash(self.tokens)
-    value = (value * 31) ^ hash(self.entityType)
-    value = (value * 31) ^ hash(self.phraseType)
-    value = (value * 31) ^ hash(self.confidence)
-    value = (value * 31) ^ hash(self.text)
-    value = (value * 31) ^ hash(self.childMentionIdList)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -625,14 +596,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.uuid)
-    value = (value * 31) ^ hash(self.metadata)
-    value = (value * 31) ^ hash(self.mentionList)
-    value = (value * 31) ^ hash(self.linkingList)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
diff -ru a/concrete/exceptions/ttypes.py b/concrete/exceptions/ttypes.py
--- a/concrete/exceptions/ttypes.py	2015-09-26 07:59:51.396064000 -0400
+++ b/concrete/exceptions/ttypes.py	2015-09-26 07:29:59.304051000 -0400
@@ -83,12 +83,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.message)
-    value = (value * 31) ^ hash(self.serEx)
-    return value
-
   def __str__(self):
     return repr(self)
 
diff -ru a/concrete/__init__.py b/concrete/__init__.py
--- a/concrete/__init__.py	2015-09-26 07:59:51.167217000 -0400
+++ b/concrete/__init__.py	2015-09-26 07:29:58.547078000 -0400
@@ -1,27 +0,0 @@
-# This file is NOT auto-generated
-
-"""
-Python modules and scripts for working with
-[Concrete](https://github.com/hltcoe/concrete), an HLT data
-specification defined using [Thrift](http://thrift.apache.org).
-"""
-
-
-from audio.ttypes import *
-from clustering.ttypes import *
-from communication.ttypes import *
-from communication_service.ttypes import *
-from discourse.ttypes import *
-from email.ttypes import *
-from entities.ttypes import *
-from language.ttypes import *
-from linking.ttypes import *
-from metadata.ttypes import *
-from nitf.ttypes import *
-from situations.ttypes import *
-from spans.ttypes import *
-from structure.ttypes import *
-from twitter.ttypes import *
-from uuid.ttypes import *
-
-from version import __version__
Only in a/concrete: inspect.py
diff -ru a/concrete/language/ttypes.py b/concrete/language/ttypes.py
--- a/concrete/language/ttypes.py	2015-09-26 07:59:51.061919000 -0400
+++ b/concrete/language/ttypes.py	2015-09-26 07:29:58.838555000 -0400
@@ -125,13 +125,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.uuid)
-    value = (value * 31) ^ hash(self.metadata)
-    value = (value * 31) ^ hash(self.languageToProbabilityMap)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
diff -ru a/concrete/linking/ttypes.py b/concrete/linking/ttypes.py
--- a/concrete/linking/ttypes.py	2015-09-26 07:59:51.266739000 -0400
+++ b/concrete/linking/ttypes.py	2015-09-26 07:29:58.875565000 -0400
@@ -121,14 +121,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.confidence)
-    value = (value * 31) ^ hash(self.targetId)
-    value = (value * 31) ^ hash(self.dbId)
-    value = (value * 31) ^ hash(self.dbName)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -217,12 +209,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.sourceId)
-    value = (value * 31) ^ hash(self.linkTargetList)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -311,12 +297,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.metadata)
-    value = (value * 31) ^ hash(self.linkList)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
diff -ru a/concrete/metadata/ttypes.py b/concrete/metadata/ttypes.py
--- a/concrete/metadata/ttypes.py	2015-09-26 07:59:51.157370000 -0400
+++ b/concrete/metadata/ttypes.py	2015-09-26 07:29:58.914118000 -0400
@@ -401,25 +401,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.sectionTheoryList)
-    value = (value * 31) ^ hash(self.sentenceTheoryList)
-    value = (value * 31) ^ hash(self.tokenizationTheoryList)
-    value = (value * 31) ^ hash(self.posTagTheoryList)
-    value = (value * 31) ^ hash(self.nerTagTheoryList)
-    value = (value * 31) ^ hash(self.lemmaTheoryList)
-    value = (value * 31) ^ hash(self.langIdTheoryList)
-    value = (value * 31) ^ hash(self.parseTheoryList)
-    value = (value * 31) ^ hash(self.dependencyParseTheoryList)
-    value = (value * 31) ^ hash(self.tokenAnnotationTheoryList)
-    value = (value * 31) ^ hash(self.entityMentionSetTheoryList)
-    value = (value * 31) ^ hash(self.entitySetTheoryList)
-    value = (value * 31) ^ hash(self.situationMentionSetTheoryList)
-    value = (value * 31) ^ hash(self.situationSetTheoryList)
-    value = (value * 31) ^ hash(self.communicationsList)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -585,17 +566,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.bytesValue)
-    value = (value * 31) ^ hash(self.int64Value)
-    value = (value * 31) ^ hash(self.doubleValue)
-    value = (value * 31) ^ hash(self.stringValue)
-    value = (value * 31) ^ hash(self.int64List)
-    value = (value * 31) ^ hash(self.doubleList)
-    value = (value * 31) ^ hash(self.stringList)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -725,15 +695,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.tool)
-    value = (value * 31) ^ hash(self.timestamp)
-    value = (value * 31) ^ hash(self.digest)
-    value = (value * 31) ^ hash(self.dependencies)
-    value = (value * 31) ^ hash(self.kBest)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -756,7 +717,7 @@
    - tweetInfo: Extra information for communications where kind==TWEET:
   Information about this tweet that is provided by the Twitter
   API.  For information about the Twitter API, see:
-  <https://dev.twitter.com/docs/platform-objects>
+  https://dev.twitter.com/docs/platform-objects
    - emailInfo: Extra information for communications where kind==EMAIL
    - nitfInfo: Extra information that may come from the NITF
   (News Industry Text Format) schema. See 'nitf.thrift'.
@@ -830,13 +791,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.tweetInfo)
-    value = (value * 31) ^ hash(self.emailInfo)
-    value = (value * 31) ^ hash(self.nitfInfo)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
diff -ru a/concrete/nitf/ttypes.py b/concrete/nitf/ttypes.py
--- a/concrete/nitf/ttypes.py	2015-09-26 07:59:51.012983000 -0400
+++ b/concrete/nitf/ttypes.py	2015-09-26 07:29:58.969013000 -0400
@@ -1,9 +1,3 @@
-# -*- coding: utf-8 -*-
-
-# The 'coding' line above was manually added to the autogenerated code
-# to prevent the error message on import:
-#   SyntaxError: Non-ASCII character '\xef' in file concrete/nitf/ttypes.py
-
 #
 # Autogenerated by Thrift Compiler (0.9.1)
 #
@@ -49,9 +43,9 @@
   </ol>
    - columnName: If the article is part of a regular column, this field specifies the name
   of that column.
-  <br />
+  <br>
   Sample Column Names:
-  <br />
+  <br>
   <ol>
   <li>World News Briefs</li>
   <li>WEDDINGS</li>
@@ -83,7 +77,7 @@
    - descriptorList: The &quot;descriptors&quot; field specifies a list of descriptive terms drawn from
   a normalized controlled vocabulary corresponding to subjects mentioned in
   the article.
-  <br />
+  <br>
   Examples Include:
   <ol>
   <li>ECONOMIC CONDITIONS AND TRENDS</li>
@@ -98,7 +92,7 @@
    - generalOnlineDescriptorList: The &quot;general online descriptors&quot; field specifies a list of descriptors
   that are at a higher level of generality than the other tags associated
   with the article.
-  <br />
+  <br>
   Examples Include:
   <ol>
   <li>Surfing</li>
@@ -115,7 +109,7 @@
    - locationList: The &quot;locations&quot; field specifies a list of geographic descriptors drawn
   from a normalized controlled vocabulary that correspond to places
   mentioned in the article.
-  <br />
+  <br>
   Examples Include:
   <ol>
   <li>Wellsboro (Pa)</li>
@@ -123,7 +117,7 @@
   <li>Park Slope (NYC)</li>
   </ol>
    - nameList: The &quot;names&quot; field specifies a list of names mentioned in the article.
-  <br />
+  <br>
   Examples Include:
   <ol>
   <li>Azza Fahmy</li>
@@ -137,7 +131,7 @@
   name, first name).
    - onlineDescriptorList: This field specifies a list of descriptors from a normalized controlled
   vocabulary that correspond to topics mentioned in the article.
-  <br />
+  <br>
   Examples Include:
   <ol>
   <li>Marriages</li>
@@ -149,7 +143,7 @@
    - onlineLeadParagraph: This field specifies the lead paragraph for the online version.
    - onlineLocationList: This field specifies a list of place names that correspond to geographic
   locations mentioned in the article.
-  <br />
+  <br>
   Examples Include:
   <ol>
   <li>Hollywood</li>
@@ -158,7 +152,7 @@
   </ol>
    - onlineOrganizationList: This field specifies a list of organizations that correspond to
   organizations mentioned in the article.
-  <br />
+  <br>
   Examples Include:
   <ol>
   <li>Nintendo Company Limited</li>
@@ -167,7 +161,7 @@
   </ol>
    - onlinePeople: This field specifies a list of people that correspond to individuals
   mentioned in the article.
-  <br/>
+  <br>
   Examples Include:
   <ol>
   <li>Lopez, Jennifer</li>
@@ -177,17 +171,17 @@
    - onlineSectionList: This field specifies the section(s) in which the online version of the article
   is placed. This may typically be populated from a semicolon (;) delineated list.
    - onlineTitleList: This field specifies a list of authored works mentioned in the article.
-  <br/>
+  <br>
   Examples Include:
   <ol>
   <li>Matchstick Men (Movie)</li>
   <li>Blades of Glory (Movie)</li>
-  <li>Bridge & Tunnel (Play)</li>
+  <li>Bridge and Tunnel (Play)</li>
   </ol>
    - organizationList: This field specifies a list of organization names drawn from a normalized
   controlled vocabulary that correspond to organizations mentioned in the
   article.
-  <br />
+  <br>
   Examples Include:
   <ol>
   <li>Circuit City Stores Inc</li>
@@ -201,7 +195,7 @@
   the <strong>section</strong> field.
    - peopleList: This field specifies a list of people from a normalized controlled
   vocabulary that correspond to individuals mentioned in the article.
-  <br />
+  <br>
   Examples Include:
   <ol>
   <li>REAGAN, RONALD WILSON (PRES)</li>
@@ -230,7 +224,7 @@
    - taxonomicClassifierList: This field specifies a list of taxonomic classifiers that place this
   article into a hierarchy of articles. The individual terms of each
   taxonomic classifier are separated with the '/' character.
-  <br/>
+  <br>
   Examples Include:
   <ol>
   <li>Top/Features/Travel/Guides/Destinations/North America/United
@@ -240,16 +234,16 @@
   </ol>
    - titleList: This field specifies a list of authored works that correspond to works
   mentioned in the article.
-  <br/>
+  <br>
   Examples Include:
   <ol>
   <li>Greystoke: The Legend of Tarzan, Lord of the Apes (Movie)</li>
-  <li>Law & Order (TV Program)</li>
+  <li>Law and Order (TV Program)</li>
   <li>BATTLEFIELD EARTH (BOOK)</li>
   </ol>
    - typesOfMaterialList: This field specifies a normalized list of terms describing the general
   editorial category of the article.
-  <br />
+  <br>
   Examples Include:
   <ol>
   <li>REVIEW</li>
@@ -915,54 +909,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.alternateURL)
-    value = (value * 31) ^ hash(self.articleAbstract)
-    value = (value * 31) ^ hash(self.authorBiography)
-    value = (value * 31) ^ hash(self.banner)
-    value = (value * 31) ^ hash(self.biographicalCategoryList)
-    value = (value * 31) ^ hash(self.columnName)
-    value = (value * 31) ^ hash(self.columnNumber)
-    value = (value * 31) ^ hash(self.correctionDate)
-    value = (value * 31) ^ hash(self.correctionText)
-    value = (value * 31) ^ hash(self.credit)
-    value = (value * 31) ^ hash(self.dayOfWeek)
-    value = (value * 31) ^ hash(self.descriptorList)
-    value = (value * 31) ^ hash(self.featurePage)
-    value = (value * 31) ^ hash(self.generalOnlineDescriptorList)
-    value = (value * 31) ^ hash(self.guid)
-    value = (value * 31) ^ hash(self.kicker)
-    value = (value * 31) ^ hash(self.leadParagraphList)
-    value = (value * 31) ^ hash(self.locationList)
-    value = (value * 31) ^ hash(self.nameList)
-    value = (value * 31) ^ hash(self.newsDesk)
-    value = (value * 31) ^ hash(self.normalizedByline)
-    value = (value * 31) ^ hash(self.onlineDescriptorList)
-    value = (value * 31) ^ hash(self.onlineHeadline)
-    value = (value * 31) ^ hash(self.onlineLeadParagraph)
-    value = (value * 31) ^ hash(self.onlineLocationList)
-    value = (value * 31) ^ hash(self.onlineOrganizationList)
-    value = (value * 31) ^ hash(self.onlinePeople)
-    value = (value * 31) ^ hash(self.onlineSectionList)
-    value = (value * 31) ^ hash(self.onlineTitleList)
-    value = (value * 31) ^ hash(self.organizationList)
-    value = (value * 31) ^ hash(self.page)
-    value = (value * 31) ^ hash(self.peopleList)
-    value = (value * 31) ^ hash(self.publicationDate)
-    value = (value * 31) ^ hash(self.publicationDayOfMonth)
-    value = (value * 31) ^ hash(self.publicationMonth)
-    value = (value * 31) ^ hash(self.publicationYear)
-    value = (value * 31) ^ hash(self.section)
-    value = (value * 31) ^ hash(self.seriesName)
-    value = (value * 31) ^ hash(self.slug)
-    value = (value * 31) ^ hash(self.taxonomicClassifierList)
-    value = (value * 31) ^ hash(self.titleList)
-    value = (value * 31) ^ hash(self.typesOfMaterialList)
-    value = (value * 31) ^ hash(self.url)
-    value = (value * 31) ^ hash(self.wordCount)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
diff -ru a/concrete/services/Annotator.py b/concrete/services/Annotator.py
--- a/concrete/services/Annotator.py	2015-09-26 07:59:51.320941000 -0400
+++ b/concrete/services/Annotator.py	2015-09-26 07:29:59.480645000 -0400
@@ -310,11 +310,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.original)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -388,12 +383,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.success)
-    value = (value * 31) ^ hash(self.ex)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -496,11 +485,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.success)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -602,11 +586,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.success)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
diff -ru a/concrete/services/__init__.py b/concrete/services/__init__.py
--- a/concrete/services/__init__.py	2015-09-26 07:59:51.324399000 -0400
+++ b/concrete/services/__init__.py	2015-09-26 07:29:59.468489000 -0400
@@ -1 +1 @@
-__all__ = ['ttypes', 'constants', 'Annotator']
+__all__ = ['ttypes', 'constants', 'Annotator', 'CommunicationService']
diff -ru a/concrete/situations/ttypes.py b/concrete/situations/ttypes.py
--- a/concrete/situations/ttypes.py	2015-09-26 07:59:51.122920000 -0400
+++ b/concrete/situations/ttypes.py	2015-09-26 07:29:59.063011000 -0400
@@ -107,13 +107,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.value)
-    value = (value * 31) ^ hash(self.metadata)
-    value = (value * 31) ^ hash(self.polarity)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -230,14 +223,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.role)
-    value = (value * 31) ^ hash(self.entityId)
-    value = (value * 31) ^ hash(self.situationId)
-    value = (value * 31) ^ hash(self.propertyList)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -340,13 +325,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.justificationType)
-    value = (value * 31) ^ hash(self.mentionId)
-    value = (value * 31) ^ hash(self.tokenRefSeqList)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -433,13 +411,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.timeMLClass)
-    value = (value * 31) ^ hash(self.timeMLTense)
-    value = (value * 31) ^ hash(self.timeMLAspect)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -871,20 +842,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.uuid)
-    value = (value * 31) ^ hash(self.situationType)
-    value = (value * 31) ^ hash(self.situationKind)
-    value = (value * 31) ^ hash(self.argumentList)
-    value = (value * 31) ^ hash(self.mentionIdList)
-    value = (value * 31) ^ hash(self.justificationList)
-    value = (value * 31) ^ hash(self.timeML)
-    value = (value * 31) ^ hash(self.intensity)
-    value = (value * 31) ^ hash(self.polarity)
-    value = (value * 31) ^ hash(self.confidence)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -1010,14 +967,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.uuid)
-    value = (value * 31) ^ hash(self.metadata)
-    value = (value * 31) ^ hash(self.situationList)
-    value = (value * 31) ^ hash(self.linkingList)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -1175,17 +1124,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.role)
-    value = (value * 31) ^ hash(self.entityMentionId)
-    value = (value * 31) ^ hash(self.situationMentionId)
-    value = (value * 31) ^ hash(self.tokens)
-    value = (value * 31) ^ hash(self.confidence)
-    value = (value * 31) ^ hash(self.propertyList)
-    value = (value * 31) ^ hash(self.constituent)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -1600,20 +1538,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.uuid)
-    value = (value * 31) ^ hash(self.text)
-    value = (value * 31) ^ hash(self.situationType)
-    value = (value * 31) ^ hash(self.situationKind)
-    value = (value * 31) ^ hash(self.argumentList)
-    value = (value * 31) ^ hash(self.intensity)
-    value = (value * 31) ^ hash(self.polarity)
-    value = (value * 31) ^ hash(self.tokens)
-    value = (value * 31) ^ hash(self.constituent)
-    value = (value * 31) ^ hash(self.confidence)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -1739,14 +1663,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.uuid)
-    value = (value * 31) ^ hash(self.metadata)
-    value = (value * 31) ^ hash(self.mentionList)
-    value = (value * 31) ^ hash(self.linkingList)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
diff -ru a/concrete/spans/ttypes.py b/concrete/spans/ttypes.py
--- a/concrete/spans/ttypes.py	2015-09-26 07:59:51.145118000 -0400
+++ b/concrete/spans/ttypes.py	2015-09-26 07:29:59.097448000 -0400
@@ -94,12 +94,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.start)
-    value = (value * 31) ^ hash(self.ending)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -184,12 +178,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.start)
-    value = (value * 31) ^ hash(self.ending)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
diff -ru a/concrete/structure/ttypes.py b/concrete/structure/ttypes.py
--- a/concrete/structure/ttypes.py	2015-09-26 07:59:50.967554000 -0400
+++ b/concrete/structure/ttypes.py	2015-09-26 07:29:59.171664000 -0400
@@ -181,15 +181,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.tokenIndex)
-    value = (value * 31) ^ hash(self.text)
-    value = (value * 31) ^ hash(self.textSpan)
-    value = (value * 31) ^ hash(self.rawTextSpan)
-    value = (value * 31) ^ hash(self.audioSpan)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -269,12 +260,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.parseId)
-    value = (value * 31) ^ hash(self.constituentIndex)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -440,16 +425,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.tokenIndexList)
-    value = (value * 31) ^ hash(self.anchorTokenIndex)
-    value = (value * 31) ^ hash(self.tokenizationId)
-    value = (value * 31) ^ hash(self.textSpan)
-    value = (value * 31) ^ hash(self.rawTextSpan)
-    value = (value * 31) ^ hash(self.audioSpan)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -586,15 +561,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.tokenIndex)
-    value = (value * 31) ^ hash(self.tag)
-    value = (value * 31) ^ hash(self.confidence)
-    value = (value * 31) ^ hash(self.tagList)
-    value = (value * 31) ^ hash(self.confidenceList)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -731,14 +697,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.uuid)
-    value = (value * 31) ^ hash(self.metadata)
-    value = (value * 31) ^ hash(self.taggedTokenList)
-    value = (value * 31) ^ hash(self.taggingType)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -827,13 +785,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.gov)
-    value = (value * 31) ^ hash(self.dep)
-    value = (value * 31) ^ hash(self.edgeType)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -944,14 +895,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.isAcyclic)
-    value = (value * 31) ^ hash(self.isConnected)
-    value = (value * 31) ^ hash(self.isSingleHeaded)
-    value = (value * 31) ^ hash(self.isProjective)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -1068,14 +1011,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.uuid)
-    value = (value * 31) ^ hash(self.metadata)
-    value = (value * 31) ^ hash(self.dependencyList)
-    value = (value * 31) ^ hash(self.structureInformation)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -1219,16 +1154,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.id)
-    value = (value * 31) ^ hash(self.tag)
-    value = (value * 31) ^ hash(self.childList)
-    value = (value * 31) ^ hash(self.headChildIndex)
-    value = (value * 31) ^ hash(self.start)
-    value = (value * 31) ^ hash(self.ending)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -1339,13 +1264,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.uuid)
-    value = (value * 31) ^ hash(self.metadata)
-    value = (value * 31) ^ hash(self.constituentList)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -1429,12 +1347,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.weight)
-    value = (value * 31) ^ hash(self.tokenList)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -1534,14 +1446,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.src)
-    value = (value * 31) ^ hash(self.dst)
-    value = (value * 31) ^ hash(self.token)
-    value = (value * 31) ^ hash(self.weight)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -1674,14 +1578,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.startState)
-    value = (value * 31) ^ hash(self.endState)
-    value = (value * 31) ^ hash(self.arcList)
-    value = (value * 31) ^ hash(self.cachedBestPath)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -1755,11 +1651,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.tokenList)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -1868,14 +1759,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.tokens)
-    value = (value * 31) ^ hash(self.concreteTarget)
-    value = (value * 31) ^ hash(self.externalTarget)
-    value = (value * 31) ^ hash(self.linkType)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -2113,28 +1996,9 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.uuid)
-    value = (value * 31) ^ hash(self.metadata)
-    value = (value * 31) ^ hash(self.tokenList)
-    value = (value * 31) ^ hash(self.lattice)
-    value = (value * 31) ^ hash(self.kind)
-    value = (value * 31) ^ hash(self.tokenTaggingList)
-    value = (value * 31) ^ hash(self.parseList)
-    value = (value * 31) ^ hash(self.dependencyParseList)
-    value = (value * 31) ^ hash(self.spanLinkList)
-    return value
-
   def __repr__(self):
-    # The implementation of this function is NOT auto-generated
-    # Prevent infinite recursion caused by backpointer added by
-    #   concrete.util.references.add_references_to_communication()
-    dict_without_backpointer = self.__dict__.copy()
-    if 'sentence' in dict_without_backpointer:
-      dict_without_backpointer.pop('sentence')
     L = ['%s=%r' % (key, value)
-      for key, value in dict_without_backpointer.iteritems()]
+      for key, value in self.__dict__.iteritems()]
     return '%s(%s)' % (self.__class__.__name__, ', '.join(L))
 
   def __eq__(self, other):
@@ -2271,15 +2135,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.uuid)
-    value = (value * 31) ^ hash(self.tokenization)
-    value = (value * 31) ^ hash(self.textSpan)
-    value = (value * 31) ^ hash(self.rawTextSpan)
-    value = (value * 31) ^ hash(self.audioSpan)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -2505,19 +2360,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.uuid)
-    value = (value * 31) ^ hash(self.sentenceList)
-    value = (value * 31) ^ hash(self.textSpan)
-    value = (value * 31) ^ hash(self.rawTextSpan)
-    value = (value * 31) ^ hash(self.audioSpan)
-    value = (value * 31) ^ hash(self.kind)
-    value = (value * 31) ^ hash(self.label)
-    value = (value * 31) ^ hash(self.numberList)
-    value = (value * 31) ^ hash(self.lidList)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
diff -ru a/concrete/twitter/ttypes.py b/concrete/twitter/ttypes.py
--- a/concrete/twitter/ttypes.py	2015-09-26 07:59:50.799934000 -0400
+++ b/concrete/twitter/ttypes.py	2015-09-26 07:29:59.243093000 -0400
@@ -262,27 +262,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.id)
-    value = (value * 31) ^ hash(self.name)
-    value = (value * 31) ^ hash(self.screenName)
-    value = (value * 31) ^ hash(self.lang)
-    value = (value * 31) ^ hash(self.geoEnabled)
-    value = (value * 31) ^ hash(self.createdAt)
-    value = (value * 31) ^ hash(self.friendsCount)
-    value = (value * 31) ^ hash(self.statusesCount)
-    value = (value * 31) ^ hash(self.verified)
-    value = (value * 31) ^ hash(self.listedCount)
-    value = (value * 31) ^ hash(self.favouritesCount)
-    value = (value * 31) ^ hash(self.followersCount)
-    value = (value * 31) ^ hash(self.location)
-    value = (value * 31) ^ hash(self.timeZone)
-    value = (value * 31) ^ hash(self.description)
-    value = (value * 31) ^ hash(self.utcOffset)
-    value = (value * 31) ^ hash(self.url)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -357,12 +336,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.latitude)
-    value = (value * 31) ^ hash(self.longitude)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -446,12 +419,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.type)
-    value = (value * 31) ^ hash(self.coordinateList)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -536,13 +503,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.streetAddress)
-    value = (value * 31) ^ hash(self.region)
-    value = (value * 31) ^ hash(self.locality)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -652,15 +612,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.startOffset)
-    value = (value * 31) ^ hash(self.endOffset)
-    value = (value * 31) ^ hash(self.screenName)
-    value = (value * 31) ^ hash(self.name)
-    value = (value * 31) ^ hash(self.id)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -769,15 +720,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.startOffset)
-    value = (value * 31) ^ hash(self.endOffset)
-    value = (value * 31) ^ hash(self.expandedUrl)
-    value = (value * 31) ^ hash(self.url)
-    value = (value * 31) ^ hash(self.displayUrl)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -862,13 +804,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.text)
-    value = (value * 31) ^ hash(self.startOffset)
-    value = (value * 31) ^ hash(self.endOffset)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -980,13 +915,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.hashtagList)
-    value = (value * 31) ^ hash(self.urlList)
-    value = (value * 31) ^ hash(self.userMentionList)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -1145,19 +1073,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.placeType)
-    value = (value * 31) ^ hash(self.countryCode)
-    value = (value * 31) ^ hash(self.country)
-    value = (value * 31) ^ hash(self.fullName)
-    value = (value * 31) ^ hash(self.name)
-    value = (value * 31) ^ hash(self.id)
-    value = (value * 31) ^ hash(self.url)
-    value = (value * 31) ^ hash(self.boundingBox)
-    value = (value * 31) ^ hash(self.attributes)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -1231,12 +1146,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.type)
-    value = (value * 31) ^ hash(self.coordinates)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
@@ -1472,25 +1381,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.id)
-    value = (value * 31) ^ hash(self.text)
-    value = (value * 31) ^ hash(self.createdAt)
-    value = (value * 31) ^ hash(self.user)
-    value = (value * 31) ^ hash(self.truncated)
-    value = (value * 31) ^ hash(self.entities)
-    value = (value * 31) ^ hash(self.source)
-    value = (value * 31) ^ hash(self.coordinates)
-    value = (value * 31) ^ hash(self.place)
-    value = (value * 31) ^ hash(self.favorited)
-    value = (value * 31) ^ hash(self.retweeted)
-    value = (value * 31) ^ hash(self.retweetCount)
-    value = (value * 31) ^ hash(self.inReplyToScreenName)
-    value = (value * 31) ^ hash(self.inReplyToStatusId)
-    value = (value * 31) ^ hash(self.inReplyToUserId)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
Only in a/concrete: util
diff -ru a/concrete/uuid/ttypes.py b/concrete/uuid/ttypes.py
--- a/concrete/uuid/ttypes.py	2015-09-26 07:59:50.693904000 -0400
+++ b/concrete/uuid/ttypes.py	2015-09-26 07:29:59.271810000 -0400
@@ -71,11 +71,6 @@
     return
 
 
-  def __hash__(self):
-    value = 17
-    value = (value * 31) ^ hash(self.uuidString)
-    return value
-
   def __repr__(self):
     L = ['%s=%r' % (key, value)
       for key, value in self.__dict__.iteritems()]
Only in a/concrete: validate.py
Only in a/concrete: version.py
